src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -598,15 +750,74 @@ void MPS::full_state_vector(cvector_t& statevector) const
 #endif
 }
 
-void MPS::probabilities_vector(rvector_t& probvector) const
+void MPS::probabilities_vector(rvector_t& probvector, 
+			       const reg_t &qubits) const
 {
-  MPS_Tensor mps_vec = state_vec(0, num_qubits_-1);
-  uint_t length = 1ULL << num_qubits_;   // length = pow(2, num_qubits_)
+  cvector_t state_vec;
+  uint_t num_qubits = qubits.size();
+  uint_t length = 1ULL << num_qubits;   // length = pow(2, num_qubits)
   probvector.resize(length);
-  #pragma omp parallel for
-  for (int_t i = 0; i < static_cast<int_t>(length); i++) {
-    probvector[i] = std::norm(mps_vec.get_data(reverse_bits(i, num_qubits_))(0,0));
+
+  bool ordered = true;
+  for (uint_t index=0; index < num_qubits-1; index++) {
+    if (qubits[index] > qubits[index+1]){
+      ordered = false;
+      break;
+    }
+  }
+  bool reverse_ordered = true;
+  for (uint_t index=0; index < num_qubits-1; index++) {
+    if (qubits[index] < qubits[index+1]){
```

Comment:
Don't we just need to test one case to know whether is reversed or not?
like:
```c++
    reversed = (qubits[0] < qubits[1] ? false: true);
```

Reply:
No. I wanted to check it **all** qubits are ordered, i.e., `qubits[i]<qubits[i+1]` for `all i`. 
Or if all qubits are in reverse order, i.e., `qubits[i]>qubits[i+1]` for `all i`.
I wrote these special cases initially as an optimization, in order not to need to compute the density matrix. But after looking at it again because of your comment, I realized I could remove all those special cases, because I only compute the **trace** of the density matrix anyway, so the special cases don't improve performance. 
In short, I rewrote the function and now it is a lot shorter and simpler.